packages/plugins/typescript/react-query/src/config.ts

```diff
@@ -115,6 +115,12 @@ export interface ReactQueryRawPluginConfig
    */
   legacyMode?: boolean;
 
+  /**
+   * @default 4
+   * @description The version of react-query to use. Default is 4. legacyMode must be false to active this flag.
+   */
+  reactQueryVersion?: 4 | 5;
```

Comment:
💬 suggestion: I'd unify this to `reactQueryVersion?: 3 | 4 | 5`. 3 would be legacy.

this would be a breaking change, but it would be better from an api perspective

Reply:
@TkDodo Do you mean remove the legacyMode flag and unify the reactQueryVersion?: 3 | 4 | 5 ? Do you suggest to keep both and have a waring info for preventing breaking change?

Reply:
if you want to avoid breaking changes, keeping the `legacyMode` and deprecating it is one option. This is more something for the maintainers of this package to decide though.